tcms/telemetry/api.py

```diff
@@ -0,0 +1,45 @@
+from django.db.models import Count
+from modernrpc.core import rpc_method
+
+from tcms.testcases.models import TestCase
+
+
+@rpc_method(name='Testing.breakdown')
+def breakdown(query=None):
+    """
+    .. function:: XML-RPC Testing.breakdown(query)
+
+        Perform a search and return the statistics for the selected test cases
+
+        :param query: Field lookups for :class:`tcms.testcases.models.TestCase`
+        :type query: dict
+        :return: Object, containing the statistics for the selected test cases
+        :rtype: dict
+    """
+
+    if query is None:
+        query = {}
+
+    test_cases = TestCase.objects.filter(**query).filter()
+
+    manual_count = test_cases.filter(is_automated=False).count()
+    automated_count = test_cases.filter(is_automated=True).count()
+    count = {
+        'manual': manual_count,
+        'automated': automated_count,
+        'all': manual_count + automated_count
+    }
+
+    priorities = _get_field_count_map(test_cases, 'priority', 'priority__value')
+    categories = _get_field_count_map(test_cases, 'category', 'category__name')
```

Comment:
Yes, it will look something like this:
```python
test_cases.values('priority__value', 'category__name').annotate(priority_count=Count('priority'), category_count=Count('category'))
```
however, there is a bug in this.
my current query returns the following results:
```
[['P1', 1], ['P2', 3], ['P3', 1]]
[['--default--', 4], ['other', 1]]
```
while this one returns:
```
[['P1', 1], ['P2', 2], ['P2', 1], ['P3', 1]]
[['--default--', 1], ['--default--', 2], ['other', 1], ['--default--', 1]]
```
that is because of the multiple arguments passed to `.values()`

There are 3 things I can do:
1. leave it as it is and decide whether we want to refactor it later (we are still not sure which one is the fastest)
2. come up with a smarter way to query the data
3. use the query I mentioned and aggregate the results with python code (least preferable IMO)